src/backend/access/appendonly/appendonlywriter.c

```diff
@@ -1641,6 +1683,40 @@ UpdateMasterAosegTotalsFromSegments(Relation parentrel,
 		}
 	}
 
+	if (appendonly_compaction_segno != NIL)
+	{
+		AORelHashEntryData *aoentry;
+
+		acquire_lightweight_lock();
+
+		aoentry = AORelGetOrCreateHashEntry(RelationGetRelid(parentrel));
+		Assert(aoentry);
+
+		foreach(l, appendonly_compaction_segno)
+		{
+			int			qe_segno = lfirst_int(l);
+
+			Assert(qe_segno >= 0);
+
+			if (!awaiting_drop[qe_segno])
+			{
+				AOSegfileStatus *segfilestat = &aoentry->relsegfiles[qe_segno];
+
+				elogif(Debug_appendonly_print_segfile_choice, LOG,
+					   "Update the file segment state after compaction: "
+					   "relation \"%s\" (%d), segno %d",
+					   get_rel_name(RelationGetRelid(parentrel)),
+					   RelationGetRelid(parentrel),
+					   qe_segno);
+
+				segfilestat->state = AVAILABLE;
```

Comment:
Let's remove the redundant segfilestat variable:
```
aoentry->relsegfiles[qe_segno].state = AVAILABLE;
```

Reply:
done